File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    void dfs(TreeNode node, int k, int ans[], int level[]) {
        if (node == null)
            return;

        dfs(node.left, k, ans, level);

        if (level[0]++ == k - 1) {
            ans[0] = node.val;
            return;
        }
        dfs(node.right, k, ans, level);
    }

    public int kthSmallest(TreeNode root, int k) {
        int ans[] = new int[1];
        int level[] = new int[1];

        dfs(root, k, ans, level);
        return ans[0];
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func rightSideView(root *TreeNode) []int {
    
	var list []int

	var solve func(root *TreeNode, level int)

	solve = func(root *TreeNode, level int) {

		if root==nil {
			return
		}

		if len(list)==level {
			list = append(list,root.Val)
		}

		solve(root.Right,level+1)
		solve(root.Left,level+1)
	}

	solve(root,0)
	return list

}
